Source/ASCollectionView.mm

```diff
@@ -1467,6 +1486,26 @@ - (void)_beginBatchFetching
   }
 }
 
+#pragma mark - ASCollectionViewLayoutCalculatorProviding
+
+- (id<ASCollectionViewLayoutCalculating>)collectionViewLayoutCalculator
+{
+  return _collectionViewLayoutFlags.providesLayoutCalculator ? [(id<ASCollectionViewLayoutCalculatorProviding>)self.collectionViewLayout collectionViewLayoutCalculator] : nil;
+}
+
+- (CGSize)sizeForCalculatingCollectionViewLayout
+{
+  return _collectionViewLayoutFlags.providesLayoutCalculator ? [(id<ASCollectionViewLayoutCalculatorProviding>)self.collectionViewLayout sizeForCalculatingCollectionViewLayout] : CGSizeZero;
+}
+
+#pragma mark - ASCollectionViewLayoutCalculatorResultConsuming
```

Comment:
The specificity of this name is nice, but it remains pretty abstract as to the purpose / function of this protocol.

Should we drop "View" from all of these names, even though it would be less similar to UICollectionView names?  Any other ideas on how to either simplify, or clarify the names?

This is mostly a question for the other reviewers and we should definitely not bog down this work in naming choice, but we're much less likely to revisit them after landing, so it's a good time to give it some thought.